### PR TITLE
Support `memory()` on macOS

### LINKED ISSUE #73: Can't get memory usage on macOS
This code : 

```
let sys = System::new();
let memory_usage = sys.memory().unwrap()
```
Panic with this error message : 

    thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "Not supported" }', src/main.rs:36:24

### PR DESCRIPTION
**Pull request by davidkna**
Closes #73

### CODE REVIEW

File: src/data.rs [hunk 1]
```diff
@@ -220,11 +220,12 @@ pub struct PlatformMemory {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Debug, Clone)]
 pub struct PlatformMemory {
```
Comment (davidkna) [inline src/data.rs:222]: This doesn't cover all 24 items in [`vm_statistics64`](https://docs.rs/libc/latest/x86_64-apple-darwin/libc/struct.vm_statistics64.html). Does it need to?

Comment (unrelentingtech) [reply src/data.rs:222]: I don't think so? That struct seems to include a lot of… well, statistics — like faults/pageins/pageouts whatever that are not related to *current usage*.

Comment (davidkna) [reply src/data.rs:222]: Ok. I've added most of the types that look like they are not statistics-oriented.

**Comment by unrelentingtech** [comment]
Thanks!

**Pull request merged**